vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/BaseUIUnitTest.java

```diff
@@ -113,16 +120,19 @@ private static Map<Class<?>, Class<? extends ComponentWrap>> scanWrappers(
         }
     }
 
-    static synchronized Routes discoverRoutes(String packageName) {
-        packageName = packageName == null ? "" : packageName;
-        return routesCache.computeIfAbsent(packageName,
-                pn -> new Routes().autoDiscoverViews(pn));
+    synchronized Routes discoverRoutes() {
+        return discoverRoutes(scanPackages());
+    }
+
+    static synchronized Routes discoverRoutes(Set<String> packageNames) {
+        packageNames = packageNames == null ? Set.of() : packageNames;
+        return routesCache.computeIfAbsent(packageNames, pn -> new Routes()
+                .autoDiscoverViews(pn.toArray(String[]::new)));
```

Comment:
Instead of changing the hash to depend on a set we should just do the compute if absent for each packagename and add build up a Routes object from each of them.
Else the caching is a bit too specific to be of use for multiple tests with different packages.

Reply:
Changed by adding a `merge` operation to `Routes` that makes a copy with data from two other `Routes`.
Did not use the constructor directly because it takes a mutable sets and passing the ones from the source route may corrupt them with unwanted data.